lib/Sat/CryptoMinisat5.cpp

```diff
@@ -90,11 +95,20 @@ bool CryptoMiniSat5::solve(bool& timeout_expired) // Search without assumptions.
      s->set_max_confl(std::max(max_confl - s->get_sum_conflicts(), (uint64_t)1));
   }
 
+  /*
+   * STP uses -1 for a value of "no timeout" -- this means that we only set the
+   * timeout _in the SAT solver_ if the value is >= 0. This avoids us
+   * accidentally setting a large limit (or one in the past).
+   */
+  if (max_time > 0) {
+     s->set_max_time(max_time);
+  }
+
   CMSat::lbool ret = s->solve();
   if (ret == CMSat::l_Undef)
   {
     timeout_expired = true;
-    assert(s->get_sum_conflicts() >= max_confl);
+    // TODO: do we want an assertion on the case of a timeout?
```

Comment:
No, **definitely** not. Please remove this TODO -- it should not be there. The time can run while e.g. someone had a debug point set or the time can be weirdly linear and even non-linear (think of summer time, winter time, people changing the time by hand, etc). So time is VERY weird and we should definitely not assert-fail because of that.

Reply:
Done!